src/main/java/fi/aalto/cs/apluscourses/ui/base/TreeView.java

```diff
@@ -48,7 +50,7 @@
     return (SelectableNodeViewModel<?>) TREE_MODEL_BUILDER.getUserObject(node);
   }
 
-  private BaseTreeViewModel<?> viewModel = null;
+  protected BaseTreeViewModel<?> viewModel = null;
```

Comment:
We don't want to set this field from the subclasses, do we? So perhaps just adding a getter to it makes more sense than changing the visibility of the field.